src/cmd/linuxkit/pkglib/docker.go

```diff
@@ -214,7 +215,7 @@ func (dr *dockerRunnerImpl) contextSupportCheck() error {
 // 1. if dockerContext is provided, try to create a builder with that context; if it succeeds, we are done; if not, return an error.
 // 2. try to find an existing named runner with the pattern; if it succeeds, we are done; if not, try next.
 // 3. try to create a generic builder using the default context named "linuxkit".
-func (dr *dockerRunnerImpl) builder(ctx context.Context, dockerContext, builderImage, platform string, restart bool) (*buildkitClient.Client, error) {
+func (dr *dockerRunnerImpl) builder(ctx context.Context, dockerContext, builderImage, platform string, restart bool) (interface{}, error) {
```

Comment:
Why is this changed to return an `interface{}`? That is about as generic as it gets. Is there any reason it cannot return a `*buildkitClient.Client`?

Reply:
Well, you are right, we have only one builder now, let's think about abstractions later.